compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -330,6 +325,12 @@ static void addGPUBufferizePasses(OpPassManager &funcPassManager) {
   funcPassManager.addPass(createCSEPass());
 }
 
+void addGPUIGEMMTileAndFusePassPipeline(
+    OpPassManager &funcPassManager, const GPUPipelineOptions &pipelineOptions) {
+  funcPassManager.addPass(createConvolutionToIGEMMPass());
+  addGPUTileAndFusePassPipeline(funcPassManager, pipelineOptions);
```

Comment:
We don't need a whole new pipeline for this, we can just call `ConvolutionToIGEMM` right at the start of `TileAndFuse` and we can control the application of im2col either with a pipeline option (probably not best but basically what this is) or with a lowering config attribute.

If you want to start with a pipeline option that works for me though.

Reply:
I started with the pipeline option, but then ran into all the python binding stuff that was recently added, and decided to just add a new pipeline. The nice thing about making it a new pass/pipeline option is that we can get rid of the control function, and the wrapper pass for LLVMGPU, so I don't really want to make it a lowering config attribute.

Reply:
I've been trying to reduce the number of pipelines as pipelines have high maintenance cost and are much harder to configure as we try to fuse more both horizontally and vertically.

I see that this is basically a wrapper pipeline but we should be controlling applications of individual passes with pipeline options rather than new entry points. We've been planning to deduplicate LLVMGPUPadAndVectorDistribute for this reason for a while, and the rest of the compiler (Flow/Stream/HAL) all do this already.

FWIW since igemm is an algorithmic change that data seems to be showing we might not always want to do, I'm expecting we'll land on lowering config eventually anyway but I'm not blocking on that.